PatternPal/PatternPal.Extension/Commands/SubscribeEvents.cs

```diff
@@ -0,0 +1,179 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+using PatternPal.Extension.Commands;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Grpc.Core;
+using PatternPal.Extension.Grpc;
+using StreamJsonRpc;
+using EnvDTE;
+using PatternPal.Extension.UserControls;
+using PatternPal.Extension.ViewModels;
+using EnvDTE80;
+using Microsoft.Win32;
+using PatternPal.Protos;
+using PatternPal.Extension.Model;
+
+namespace PatternPal.Extension.Commands
+{
+    public static class SubscribeEvents
+    {
+        /// <summary>
+        ///     VS Package that provides this command, not null.
+        /// </summary>
+        private static PatternPalExtensionPackage _package;
+
+        private static DTE _dte;
+
+        private static string _sessionId;
+
+        private static string _pathToFolder;
+        public static void Initialize(  
+            DTE dte,
+            PatternPalExtensionPackage package)
+        {
+            _dte = dte;
+            ThreadHelper.ThrowIfNotOnUIThread();
+            _package = package;
+            _pathToFolder = Path.Combine(_package.UserLocalDataPath.ToString(), "Extensions", "Team PatternPal",
+                "PatternPal.Extension");
+            SetSubjectId();
+            OnSessionStart();
+            _dte.Events.BuildEvents.OnBuildDone += OnBuildDone;
+        }
+
+        private static void OnBuildDone(
+            vsBuildScope Scope,
+            vsBuildAction Action)
+        {
+            if (!_package.DoLogData)
```

Comment:
I won't block over this now, but I think that it would make more sense if this check happens when registering the event handlers, instead of inside every event handler. That would require adding/removing the event handlers if the value of `DoLogData` changes of course!